### PR TITLE
Fixed errors on product form & duplicate menu

### PR DESCRIPTION
**Pull request by aleeks**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | See down |
| Type? | bug fix |
| Category? | BO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-1634 |
| How to test? | Play with new product form! (make errors, test duplicate, deletion...) |
- Add error when create/update product with error, play with combinations!
- Make error visible (top nav in red)
- Remove duplicate action if your aren't in shop context
- Fix delete product link, duplicate product link by removing redirect_url param....

### CODE REVIEW

File: src/Adapter/Shop/Context.php [hunk 1]
```diff
@@ -73,4 +73,56 @@ public function getShopContextGroupConstant()
...
+    }
+
+    /**
+     * Get if it's a Shop context
+     *
+     * @return bool
+     */
+    public function isShopContext()
+    {
+        return \ShopCore::getContext() === $this->getShopContextConstant();
```
Comment (thierrymarianne) [inline src/Adapter/Shop/Context.php]: I believe this method doesn't require an additional method. 
You might want to compare the returned value with the constant below directly.

File: src/PrestaShopBundle/Controller/Admin/ProductController.php [hunk 2]
```diff
@@ -759,13 +773,8 @@ public function unitAction(Request $request, $action, $id)
...
 
-        // redirect after success
-        if ($request->request->has('redirect_url')) {
-            return $this->redirect($request->get('redirect_url'), 302);
-        } else {
-            $urlGenerator = $this->container->get('prestashop.core.admin.url_generator');
-            return $this->redirect($urlGenerator->generate('admin_product_catalog'), 302);
-        }
+        $urlGenerator = $this->container->get('prestashop.core.admin.url_generator');
+        return $this->redirect($urlGenerator->generate('admin_product_catalog'), 302);
```
Comment (thierrymarianne) [inline src/PrestaShopBundle/Controller/Admin/ProductController.php]: The second parameter (302) is not required as its default value is expected.

File: src/PrestaShopBundle/Controller/Admin/ProductController.php [hunk 3]
```diff
@@ -392,16 +394,48 @@ public function formAction($id, Request $request)
             ->add('step3', 'PrestaShopBundle\Form\Admin\Product\ProductQuantity')
             ->add('step4', 'PrestaShopBundle\Form\Admin\Product\ProductShipping')
             ->add('step5', 'PrestaShopBundle\Form\Admin\Product\ProductSeo')
-            ->add('step6', 'PrestaShopBundle\Form\Admin\Product\ProductOptions')
-            ->getForm();
+            ->add('step6', 'PrestaShopBundle\Form\Admin\Product\ProductOptions');
+
+        // Prepare combination form (fake but just to validate the form)
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Controller/Admin/ProductController.php]: lulz

Comment (aleeks) [inline src/PrestaShopBundle/Controller/Admin/ProductController.php]: Agree with you.. lulz :/

**Comment by mickaelandrieu** [comment]
LGTM

**Pull request merged**